src/modules/disk_io.cpp

```diff
@@ -0,0 +1,157 @@
+#include <fstream>
+#include <iomanip>
+#include <istream>
+#include <vector>
+#include <regex>
+#include <sys/types.h>
+#include <dirent.h>
+
+#include "drawtypes/label.hpp"
+#include "drawtypes/progressbar.hpp"
+#include "modules/disk_io.hpp"
+#include "utils/math.hpp"
+
+#include "modules/meta/base.inl"
+
+POLYBAR_NS
+
+namespace modules {
+  template class module<disk_io_module>;
+
+  disk_io_module::disk_io_module(const bar_settings& bar, string name_) : timer_module<disk_io_module>(bar, move(name_)) {
+    m_interval = m_conf.get<decltype(m_interval)>(name(), "interval", 1s);
+
+    m_formatter->add(DEFAULT_FORMAT, TAG_LABEL, {TAG_LABEL}/*, TAG_BAR_READ, TAG_BAR_WRITE}*/);
+    if (m_formatter->has(TAG_LABEL)) {
+      m_label = load_optional_label(m_conf, name(), TAG_LABEL, "%speed_read% Mb/s %speed_write% Mb/s");
+    }
+    m_disk_names = _get_disk_names();
+    for (auto name : m_disk_names) {
+      std::cout << "name: " << name << std::endl;
+    }
+}
+
+std::vector<std::string> disk_io_module::_get_disk_names(void) {
+  DIR *dp;
+  struct dirent *dirp;
+  std::vector<std::string> names;
+
+  if (!(dp = opendir("/sys/block/")))
+  {
+    std::cout << "Can't open /sys/block!!" << std::endl;
+    return names;
+  }
+  while ((dirp = readdir(dp))) {
+    if (dirp->d_name[0] != '.') {
+      names.push_back(std::string(dirp->d_name));
+    }
+  }
+  closedir(dp);
+  return names;
+}
+
+std::pair<unsigned long long, unsigned long long> disk_io_module::_get_disk_read_write(std::string disk_name) {
+  unsigned long long sec_read_total_new{0ULL};
+  unsigned long long sec_write_total_new{0ULL};
+  try {
+    std::ifstream disk_io_info("/sys/block/" + disk_name + "/stat");
+
+    std::string line;
+    std::getline(disk_io_info, line);
+
+    static const std::regex re{"\\s+"};
+    std::vector<std::string> items{
```

Comment:
For easier parsing (without variable length whitespaces) you could use `/proc/diskstats/` and split it into a vector with `string_util::split()`.

Reply:
the `/proc/diskstats` actually also has continuous sequence of spaces so I can't easily split it to needed values without regular expression